### PR TITLE
Add email generation

### PR DESCRIPTION
**Pull request by rotimi-babalola**
#### What does this PR do

Generates random email addresses

### CODE REVIEW

File: lib/helpers.js [hunk 1]
```diff
@@ -40,6 +40,42 @@ var Helpers = function (ngFaker) {
 
     return state;
   };
+
+  /**
+   * slugifies string
+   *
+   * @method faker.helpers.slugify
```
Comment (BolajiOlajide) [inline lib/helpers.js]: this should be `@method ngFaker.helpers.slugify`

File: lib/helpers.js [hunk 2]
```diff
@@ -40,6 +40,42 @@ var Helpers = function (ngFaker) {
...
+   */
+  self.slugify = function (string) {
+    string = string || '';
+    return string.replace(/ /g, '-').replace(/[^\w.-]+/g, '');
+  };
+
+  /**
+   * pastYear
+   *
+   * @method faker.helpers.pastYear
```
Comment (BolajiOlajide) [inline lib/helpers.js]: this should be `@method ngFaker.helpers.pastYear`

File: lib/helpers.js [hunk 3]
```diff
@@ -40,6 +40,42 @@ var Helpers = function (ngFaker) {
...
+   * @param {number} years
+   * @param {date} reference date
+   *
+   * If we ever decide to generate dates we can move this function into a date
```
Comment (BolajiOlajide) [inline lib/helpers.js]: This should be a comment in the method and not part of the doc string.

File: lib/index.js [hunk 4]
```diff
@@ -76,7 +80,7 @@ function NgFaker(options) {
             Object.defineProperty(self.definitions[definition], property, {
                 get: function () {
                     /**
-                     * first condition ensure we default to locale fallback
+                     * first condition ensures we default to locale fallback
```
Comment (BolajiOlajide) [inline lib/index.js]: English Teacher :smh:

File: tests/internet.spec.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    });
+
+    test('it should generate a valid email if parameters are passed in', function () {
+      var email = ngfaker.internet.email('john', 'doe', 'aol.com');
+      expect(EMAIL_REGEX.test(email)).toBeTruthy();
+      expect(email.indexOf('john')).toBeGreaterThanOrEqual(0);
+      expect(email.indexOf('aol.com')).toBeGreaterThanOrEqual(0);
+    });
+  });
+});
```
Comment (BolajiOlajide) [inline tests/internet.spec.js]: New line needed here

File: lib/locales/igbo/index.js [hunk 6]
```diff
@@ -6,3 +6,4 @@ ig.title = 'Igbo';
 ig.name = require('./names');
 ig.lorem = require('./lorem');
 ig.address = require('../common/address');
+ig.internet = require('../common/internet');
```
Comment (BolajiOlajide) [inline lib/locales/igbo/index.js]: Is this class going to be different per locales? @rotimi-babalola

File: lib/index.js (hunk 4 above)
Comment (rotimi-babalola) [reply lib/index.js]: 😎

File: lib/locales/igbo/index.js (hunk 6 above)
Comment (rotimi-babalola) [reply lib/locales/igbo/index.js]: I don't think so. I was uncomfortable with doing this too. Will fix

Comment (rotimi-babalola) [reply lib/locales/igbo/index.js]: Although we did something similar for `address` @BolajiOlajide 🤔

Comment (rotimi-babalola) [reply lib/locales/igbo/index.js]: I'll just import the `internet` folder directly in the internet class 😎

File: tests/internet.spec.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  });
+
+  describe('Username', function () {
+    test('it should generate a username', function () {
+      var username = ngfaker.internet.userName();
+      expect(typeof username).toEqual('string');
+      expect(username.length).toBeGreaterThan(0);
+    });
+  });
+});
```
Comment (BolajiOlajide) [inline tests/internet.spec.js]: New line needed

File: lib/helpers.js [hunk 8]
```diff
@@ -40,6 +40,45 @@ var Helpers = function (ngFaker) {
...
+   * @param {date} reference date
+   *
+   */
+  self.pastYear = function (years, refDate) {
+    /**
+     * If we ever decide to generate dates we
+     * can move this function into a date class.
+     * For now, this can remain as a helper
+     */
+    var date = (refDate) ? new Date(Date.parse(refDate)) : new Date();
```
Comment (BolajiOlajide) [inline lib/helpers.js]: What if the format of the refDate isn't what we want?
If I enter a random string then the value of date becomes `invalid date` which will have an effect on the result of the method.

You might need to had a check to ensure the type of `refDate` is date.

File: lib/internet.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+   * @param {string} lastName
+   */
+  this.userName = function (firstName, lastName) {
+    var result;
+    firstName = firstName || ngFaker.name.firstName();
+    lastName = lastName || ngFaker.name.lastName();
+    /* eslint default-case:0 */
+    /* eslint indent:0 */
+
+    // @TODO: we should probably have a helper that funkifies user names
```
Comment (BolajiOlajide) [inline lib/internet.js]: Lol @ funkifier function

File: lib/helpers.js (hunk 8 above)
Comment (rotimi-babalola) [reply lib/helpers.js]: Good spot @BolajiOlajide I'll fix that

**Pull request merged**